crates/noirc_evaluator/src/ssa_refactor/ir/instruction.rs

```diff
@@ -442,7 +448,7 @@ pub(crate) enum InstructionResultType {
     Known(Type),
 
     /// The result type of this function is unknown and separate from its operand types.
-    /// This occurs for function calls and load operations.
+    /// TODO:WHY? This occurs for function calls and load operations.
```

Comment:
Given `Instruction::Load(v0)` what is the result type? The type of `v0` is just `Type::Reference` so we cannot know since the element type is not specified.

Given `Instruction::Call { f, args }` what is the result type? The type of `f` here again is just `Type::Function` without specifying argument or return types so we cannot know.

If we flesh out `Type::Reference` or `Type::Function` to include this information we can get rid of this variant. The original reasoning was to keep types simple and trivially copyable but we've already abandoned that with `Type::Array { .. }` no longer being copyable.